### PR TITLE
[ruff] enable A001 (builtin-variable-shadowing), A002 (builtin-argument-shadowing)

### PR DESCRIPTION
**Pull request by smackesey**
## Summary & Motivation

Turn on ruff `A001`, `A002` builtin symbol shadowing rules. Replace previously enabled pylint rules.

---

## How I Tested These Changes

BK

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12889** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12889" target="_blank">Graphite</a> 
    * **PR #12912** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12912" target="_blank">Graphite</a> 
      * **PR #12913** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12913" target="_blank">Graphite</a> 
        * **PR #12890** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12890" target="_blank">Graphite</a> 
          * **PR #12891** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12891" target="_blank">Graphite</a> 
            * **PR #12892** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12892" target="_blank">Graphite</a> 
              * **PR #12900** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12900" target="_blank">Graphite</a> 
                * **PR #12901** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12901" target="_blank">Graphite</a> 
                  * **PR #12914** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12914" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12914?utm_source=stack-comment).

File: docs/sphinx/conf.py [hunk 1]
```diff
@@ -74,7 +74,7 @@
 # -- Project information -----------------------------------------------------
 
 project = "Dagster"
-copyright = "2019, Elementl, Inc"  # pylint: disable=redefined-builtin
+copyright = "2019, Elementl, Inc"  # noqa: A001
```
Comment (alangenfeld) [inline docs/sphinx/conf.py:77]: should this date be updated?

**Review by alangenfeld — APPROVED** [review]

File: docs/sphinx/conf.py (hunk 1 above)
Comment (smackesey) [reply docs/sphinx/conf.py:77]: I don't think this shows up in the docs-- I tried searching and nothing comes up. It's just some boilerplate from the initial sphinx scaffold. So IMO no need to keep it up to date.

**Comment by smackesey** [comment]
@smackesey queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12914).

**Comment by smackesey** [comment]
[Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12914) failed to merge this pull request for an unknown reason. Try your merge again, or report a bug if you see this consistently.

**Comment by smackesey** [comment]
@smackesey queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12914).

**Comment by smackesey** [comment]
[Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12914) rebased this pull request as part of a merge.

**Comment by smackesey** [comment]
@smackesey merged this pull request with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12914).

**Pull request merged**